### PR TITLE
feat: initial tooling for migrating D1 data to Dynamo

### PR DESCRIPTION
**Pull request by travis**
To run:

```
# first ensure you are logged in to wrangler with `wrangler login`
STAGE=pr194 npm run d1-dynamo-migration
```

### CODE REVIEW

**Review by Gozala — APPROVED** [review]

**Review by vasco-santos — CHANGES_REQUESTED** [review]
wrong button 😓

**Review by vasco-santos — COMMENTED** [review]
migration code looks good, a couple of extra suggestions:

1. Could we add a Github action dispatch workflow that would run this? it can receive as input the stage and is a good way to test with a environment that will be the same for both stages
2. Should we have a validation step as well that checks that all expected content is there in expected format? Maybe read again D1 and do Dynamo Gets as needed

File: tools/d1-migration/data/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+delegations.json
```
[the remark this answers is not in the document]
Comment (travis) [reply tools/d1-migration/data/.gitignore:1]: ah $TMPDIR sounds great to me!

File: tools/d1-migration/fetch-data-from-d1.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+DB_NAME="${1:-access-staging}"
+
+mkdir -p data
+echo running "wrangler d1 execute $DB_NAME --command 'SELECT * from delegations_v3' --json | jq '.[0].results' > data/delegations.json"
+wrangler d1 execute $DB_NAME --command 'SELECT * from delegations_v3' --json | jq '.[0].results' > data/delegations.json
```
[the remark this answers is not in the document]
Comment (travis) [reply tools/d1-migration/fetch-data-from-d1.sh:7]: ah it has been in my local env, which is where I've been running this so far. I'm going to look into running it in a github action, which may require this.

**Comment by travis** [comment]
> 2. Should we have a validation step as well that checks that all expected content is there in expected format? Maybe read again D1 and do Dynamo Gets as needed

ah great idea - I'll add a little script that we can run to check this

**Comment by travis** [comment]
> 1. Could we add a Github action dispatch workflow that would run this? it can receive as input the stage and is a good way to test with a environment that will be the same for both stages

Alright I spent a few hours on this today and am running into a wall - for some reason I'm having a very hard time getting `wrangler` to use the API token I've generated, despite verifying that it is being set in the environment of the command running wrangler. 

Screenshot 2023-07-12 at 5 17 37 PM

fails with no error message, despite the same API token working when I run `wrangler d1 execute access-staging --command 'SELECT * from delegations_v3' --json` locally.

I like the idea of having a dispatch action for this BUT it's not really necessary - this migration script is pretty simple and works well locally, especially now that I've removed the "save data to a tempfile" step.

@vasco-santos would you be ok with not doing this? I'd like to timebox it because it doesn't add a ton of value over just running this manually on my machine for the one time we'll ever need to run this migration.

**Comment by vasco-santos** [comment]
> @vasco-santos would you be ok with not doing this? I'd like to timebox it because it doesn't add a ton of value over just running this manually on my machine for the one time we'll ever need to run this migration.

Yes, I suggested as a way to decouple from a specific environment and anyone could run. But was expecting it to be just adding the workflow to run 

```sh
cd tools/d1-migration
chmod +x fetch-data-from-d1.sh
./fetch-data-from-d1.sh
cd ../
STAGE=pr194 node cli.js d1-migration
```

Given now context increased to also run wrangler commands that seems good

**Pull request merged**